test/e2e/mongodbtests/mongodbtests.go

```diff
@@ -37,6 +38,35 @@ func StatefulSetIsReady(mdb *mdbv1.MongoDB) func(t *testing.T) {
 	}
 }
 
+func StatefulSetHasOwnerReference(mdb *mdbv1.MongoDB) func(t *testing.T) {
+	return func(t *testing.T) {
+		sts := appsv1.StatefulSet{}
+		err := f.Global.Client.Get(context.TODO(), types.NamespacedName{Name: mdb.Name, Namespace: f.Global.OperatorNamespace}, &sts)
+		if err != nil {
+			t.Fatal(fmt.Errorf("%s", err))
+		}
+
+		ownerReferenceMdb :=
+			*metav1.NewControllerRef(mdb, schema.GroupVersionKind{
+				Group:   mdbv1.SchemeGroupVersion.Group,
+				Version: mdbv1.SchemeGroupVersion.Version,
+				Kind:    mdb.Kind,
+			})
+
+		for _, ownerReference := range sts.GetOwnerReferences() {
+			if ownerReference.APIVersion == ownerReferenceMdb.APIVersion &&
+				ownerReference.Kind == "MongoDB" &&
+				ownerReference.Name == ownerReferenceMdb.Name &&
+				ownerReference.UID == ownerReferenceMdb.UID {
```

Comment:
Rather than checking for all of the conditions in a single if statement, I think it would be more beneficial to do an individual assertion on each field. In this case, we know either all of the fields are correct, or at least one of them isn't, we won't know which one is wrong.

We could do something like 

```golang
assert.Equal(t, ownerReferenceMdb.APIVersion, ownerReference.APIVersion)
assert.Equal(t, "MongoDB", ownerReference. Kind)
...
```

The assert library will give us a more detailed breakdown of what went wrong, what do you think?

Reply:
That makes absolutely sense, I'll do it!

Reply:
Now that I think of it, since `getOwnerReferences()` return a slice and we are checking if at least one element of it its equal to `ownerReferenceMdb`, that won't work, or am I missing something? We can't assert it on one particular element.

Reply:
well in our case we know there should be exactly 1 element, so either we could pass in a slice of owner references, or assert a single element and compare index 0